sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -93,13 +95,22 @@ object IntervalUtils {
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
 
+  def fromYearMonthString(input: UTF8String): CalendarInterval = {
+    if (input == null || input.toString == null) {
+      throw new IllegalArgumentException("Interval year-month string must be not null")
```

Comment:
cast null to interval should return null, this is incorrect.

we can add `assert(input != null)`, as `Cast` should guarantee it before calling this function

Reply:
Remove， `Cast` can keep null safe here